### PR TITLE
go mod: Bump gopkg.in/k8snetworkplumbingwg/multus-cni.v4 to 4.1.1

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
Note that it pass also without it, but better to be aligned.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer** *(optional)*:

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -440,7 +436,7 @@ k8s.io/apimachinery/pkg/version
 k8s.io/apimachinery/pkg/watch
 k8s.io/apimachinery/third_party/forked/golang/json
 k8s.io/apimachinery/third_party/forked/golang/reflect
-# k8s.io/client-go v0.29.1 => k8s.io/client-go v0.29.1
+# k8s.io/client-go v1.5.2 => k8s.io/client-go v0.29.1
```
Comment (oshoval) [inline vendor/modules.txt:439]: we might want to get rid of this replace

**Comment by oshoval** [comment]
Need first to make sure how multus is deployed, if it takes latest then this PR is ok, otherwise not yet

**Comment by oshoval** [comment]
@maiqueb can you please take a look ?

**Comment by oshoval** [comment]
Please consider creating a new release before or after this PR
it might be related to the problems we see on CNAO, so worth to try (but not sure this is the reason)

**Review by maiqueb — APPROVED** [review]

**Pull request merged**